File: content-solutions/backtracking/podium-orders.js
function solution(names, k) {
  const answer = [];
  const path = [];
  const used = new Array(names.length).fill(false);
  function pick() {
    if (path.length === k) {
      answer.push([...path]);
      return;
    }
    names.forEach((name, i) => {
      if (used[i]) return;
      used[i] = true;
      path.push(name);
      pick();
      path.pop();
      used[i] = false;
    });
  }
  pick();
  return answer;
}

File: content-solutions/backtracking/podium-orders.py
def solution(names, k):
    answer = []
    path = []
    used = [False] * len(names)

    def pick():
        if len(path) == k:
            answer.append(path[:])
            return
        for i, name in enumerate(names):
            if not used[i]:
                used[i] = True
                path.append(name)
                pick()
                path.pop()
                used[i] = False

    pick()
    return answer
